io/pci/PowerNVEEH.py

```diff
@@ -0,0 +1,218 @@
+        """
+        enter_loop = True
+        num_of_miss = 0
+        num_of_hit = 0
+        while num_of_hit <= self.max_freeze:
+            for func in self.function:
+                self.log.info("Running error inject on pe %s function %s"
+                              % (self.pci_device, func))
+                if num_of_miss < 5:
+                    return_code = self.basic_eeh(self.addr, self.err, func,
+                                                 self.phb, self.pci_device)
+                    if return_code == EEH_MISS:
+                        num_of_miss += 1
+                        self.log.info("number of miss is %d"
+                                      % num_of_miss)
+                        continue
+                    else:
+                        num_of_hit += 1
+                        self.log.info("number of hit is %d"
+                                      % num_of_hit)
+                        if num_of_hit <= self.max_freeze:
+                            if not self.check_eeh_pe_recovery():
+                                self.fail("PE %s recovery failed after"
+                                          "%d EEH" % (self.pci_device,
+                                                      num_of_hit))
+                                break
+                            else:
+                                self.log.info("PE recovered successfully")
+                else:
+                    self.log.warning("EEH inject failed for 5 times with\
+                               function %s" % func)
+                    enter_loop = False
+                    break
+            if not enter_loop:
+                break
+        else:
+            if self.check_eeh_removed():
+                self.log.info("PE %s removed successfully" % self.pci_device)
+            else:
+                self.fail("PE %s not removed after max hit" % self.pci_device)
+
+    def basic_eeh(self, addr, err, func, phb, pci_device):
+
+        """
+        Injects Error, and checks for PE recovery
+        returns True, if recovery is success, else Flase
+        """
+        self.clear_dmesg_logs()
+        return_code = self.error_inject(addr, err, func, phb, pci_device)
+        if return_code != EEH_HIT:
+            self.log.info("Skipping verification, as command failed")
+        if not self.check_eeh_hit():
+            self.log.info("PE %s EEH hit failed" % self.pci_device)
+            return EEH_MISS
+        else:
+            self.log.info("PE %s EEH hit success" % self.pci_device)
+            return EEH_HIT
+
+    @classmethod
+    def error_inject(cls, addr, err, func, phb, pci_device):
+
+        """
+        Form a command to inject the error
+        """
+        cmd = "echo %s:%s:%s:0:0 > /sys/kernel/debug/powerpc/PCI%s/err_injct \
+               && lspci -ns %s" % (addr, err, func, phb, pci_device)
+        res = process.system_output(cmd, ignore_status=True, shell=True)
+        return int(res[-1])
```

Comment:
`return int(process.system_output(cmd, ignore_status=True, shell=True)[-1])  ` will be used just want to eliminate extra variable